pkg/skuba/actions/node/upgrade/apply.go

```diff
@@ -96,7 +96,9 @@ func Apply(clientSet clientset.Interface, target *deployments.Target) error {
 			if err := node.AddTargetInformationToInitConfigurationWithClusterVersion(target, initCfg, nodeVersionInfoUpdate.Update.APIServerVersion); err != nil {
 				return errors.Wrap(err, "error adding target information to init configuration")
 			}
-			kubeadm.SetContainerImagesWithClusterVersion(initCfg, nodeVersionInfoUpdate.Update.APIServerVersion)
+			if err := kubeadm.UpdateClusterConfigurationWithClusterVersion(initCfg, nodeVersionInfoUpdate.Update.APIServerVersion); err != nil {
+				return errors.Wrap(err, "error mutating the cluster configuration")
+			}
 			initCfgContents, err = kubeadmconfigutil.MarshalInitConfigurationToBytes(initCfg, schema.GroupVersion{
 				Group:   "kubeadm.k8s.io",
 				Version: "v1beta2",
```

Comment:
Does this needs to change as `Version: kubeadm.GetKubeadmApisVersion(currentClusterVersion),`?

Reply:
Updated, `kubeadm` is the first component to be upgraded that will use this configuration, so it's even safer to retrieve the version according to the "future" version of the cluster instead of the current one.